routing/cross_mwm_road_graph.cpp

```diff
@@ -3,11 +3,138 @@
 
 #include "geometry/distance_on_sphere.hpp"
 
+#include <limits>
+
 namespace
 {
+using namespace routing;
+
 inline bool IsValidEdgeWeight(EdgeWeight const & w) { return w != INVALID_EDGE_WEIGHT; }
+
+template<class CrossNodeType>
```

Comment:
The word "Type" in template parameters usually is redundant. Is It possible to rename this just to Node?